### ISSUE TITLE
ValueError: figure size must be finite not (nan, nan)

### ISSUE THREAD

**Issue by liujianxun-ict** [numbbo/coco#1708]
Hello~
When I execute the command of ‘’python -m cocopp -o  data exdata\coco\bbobdata’ I got a ValueError as follows：
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\pyplot.py", line 697, in savefig
   res = fig.savefig(*args, **kwargs)
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\figure.py", line 1814, in savefig
   self.canvas.print_figure(fname, **kwargs)
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\backend_bases.py", line 2259, in print_figure
   **kwargs)
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\backends\backend_svg.py", line 1193, in print_svg
   return self._print_svg(filename, svgwriter, **kwargs)
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\backends\backend_svg.py", line 1238, in _print_svg
   self.figure.set_dpi(72.0)
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\figure.py", line 811, in set_dpi
   self.dpi = val
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\figure.py", line 436, in _set_dpi
   self.set_size_inches(w, h, forward=forward)
 File "D:\ProgramData\Anaconda2\lib\site-packages\matplotlib\figure.py", line 733, in set_size_inches
   '({}, {})'.format(w, h))
ValueError: figure size must be finite not (nan, nan)

Why this problem occurred and how to solve it？

**Comment by nikohansen** [comment]
Hello, it is a problem which depends on the installed `matplotlib` version, we are trying to fix the problem ASAP, @islandblue ?

**Comment by nikohansen** [comment]
FTR, with the newest `matplotlib` version the problem should not appear.

**Comment by islandblue** [comment]
I've never encountered the problem. The stack trace @liujianxun-ict shared doesn't go back far enough to see where in coco the problem occurs. @nikohansen, do we have a way of reproducing this error?

**Comment by nikohansen** [comment]
I haven't tried to reproduce the error. You are quite right, it would be _really_ useful to have the trace to where the error actually occurs in the COCO code, @liujianxun-ict ?

**Comment by liujianxun-ict** [comment]
Hello~ thanks for your @nikohansen @islandblue help. The whole stack trace are as follows.
E:\coco>python -m cocopp -o data_results exdata\coco\data3-001

Post-processing (1): will generate output data in folder data_results\data3-001
  this might take several minutes.
  Data consistent according to test in consistency_check() in pproc.DataSet
Scaling figures...
Loading best algorithm data from refalgs/best2016-bbob-biobj.tar.gz ...
  Data consistent according to test in consistency_check() in pproc.DataSet
  using: C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\refalgs/best2016-bbob-biobj.tar.gz
  done (Mon Feb 05 23:03:29 2018).
  done (Mon Feb 05 23:04:00 2018).
Generating LaTeX tables...
  done (Mon Feb 05 23:04:01 2018).
ECDF graphs...
C:\Python27\lib\site-packages\matplotlib\scale.py:111: RuntimeWarning: invalid value encountered in less_equal
  out[a <= 0] = -1000
  done (Mon Feb 05 23:04:21 2018).
ECDF graphs per function...
C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\compall\pprldmany.py:321: RuntimeWarning: invalid value encountered in les
s_equal
  tmp = (x2 <= maxval)
C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\compall\pprldmany.py:323: RuntimeWarning: invalid value encountered in les
s
  x2bis = x2[y2 < y2[tmp][-1]][-1]
C:\Python27\lib\site-packages\numpy\core\_methods.py:29: RuntimeWarning: invalid value encountered in reduce
  return umr_minimum(a, axis, None, out, keepdims)
Traceback (most recent call last):
  File "C:\Python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\__main__.py", line 373, in <module>
    rungeneric.main(args)
  File "C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\rungeneric.py", line 272, in main
    + ["-o", outputdir, alg])
  File "C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\rungeneric1.py", line 429, in main
    settings=inset)
  File "C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\compall\pprldmany.py", line 468, in all_single_functions
    settings=settings)
  File "C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\compall\pprldmany.py", line 837, in main
    ppfig.save_figure(figureName, dictAlg[algorithms_with_data[0]][0].algId)
  File "C:\Users\liujianxun\AppData\Roaming\Python\Python27\site-packages\cocopp-2.1.1-py2.7.egg\cocopp\ppfig.py", line 68, in save_figure
    bbox_inches=bbox_inches_choices.get(format, None)
  File "C:\Python27\lib\site-packages\matplotlib\pyplot.py", line 701, in savefig
    res = fig.savefig(*args, **kwargs)
  File "C:\Python27\lib\site-packages\matplotlib\figure.py", line 1834, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "C:\Python27\lib\site-packages\matplotlib\backend_bases.py", line 2267, in print_figure
    **kwargs)
  File "C:\Python27\lib\site-packages\matplotlib\backends\backend_svg.py", line 1193, in print_svg
    return self._print_svg(filename, svgwriter, **kwargs)
  File "C:\Python27\lib\site-packages\matplotlib\backends\backend_svg.py", line 1238, in _print_svg
    self.figure.set_dpi(72.0)
  File "C:\Python27\lib\site-packages\matplotlib\figure.py", line 811, in set_dpi
    self.dpi = val
  File "C:\Python27\lib\site-packages\matplotlib\figure.py", line 436, in _set_dpi
    self.set_size_inches(w, h, forward=forward)
  File "C:\Python27\lib\site-packages\matplotlib\figure.py", line 733, in set_size_inches
    '({}, {})'.format(w, h))
ValueError: figure size must be finite not (nan, nan)

These above are the overall info.

**Comment by islandblue** [comment]
@liujianxun-ict, the problem is pretty elusive. Since it doesn't seem to exist in the development branch, I must ask you to consider using the development branch in your work. Also worth noting is that the development branch will be merged into the master branch shortly, so you may be inclined to switch anyway. Please let us know if for any reason you need to keep using your version, in which case we may investigate further.

**Comment by nikohansen** [comment]
FTR,
```
pprldmany.py:321: RuntimeWarning: invalid value encountered in less_equal
tmp = (x2 <= maxval)
```
is the first thing which goes wrong. Either `x2` or `maxval` is `nan`.

**Comment by islandblue** [comment]
After a rather long study, I've come up with the following information:

The problem occurs in this piece of matplotlib code:
```
        print('*'*80) # REM
        bbox_inches = kwargs.pop("bbox_inches", None)
        print('10. bbox_inches = {}\n'.format(bbox_inches)) # REM
        if bbox_inches is None:
            bbox_inches = rcParams['savefig.bbox']
            print('20. bbox_inches = {}\n'.format(bbox_inches)) # REM

        if bbox_inches:
            # call adjust_bbox to save only the given area
            if bbox_inches == "tight":
                # when bbox_inches == "tight", it saves the figure
                # twice. The first save command is just to estimate
                # the bounding box of the figure. A stringIO object is
                # used as a temporary file object, but it causes a
                # problem for some backends (ps backend with
                # usetex=True) if they expect a filename, not a
                # file-like object. As I think it is best to change
                # the backend to support file-like object, i'm going
                # to leave it as it is. However, a better solution
                # than stringIO seems to be needed. -JJL
                result = print_method(
                    io.BytesIO(),
                    dpi=dpi,
                    facecolor=facecolor,
                    edgecolor=edgecolor,
                    orientation=orientation,
                    dryrun=True,
                    **kwargs)
                renderer = self.figure._cachedRenderer
                bbox_inches = self.figure.get_tightbbox(renderer)
                print('30. bbox_inches =\n{}\n'.format(bbox_inches)) # REM

                bbox_artists = kwargs.pop("bbox_extra_artists", None)
                if bbox_artists is None:
                    bbox_artists = self.figure.get_default_bbox_extra_artists()

                bbox_filtered = []
                for a in bbox_artists:
                    print('31. a = {}'.format(a)) # REM
                    bbox = a.get_window_extent(renderer)
                    print('32. bbox = {}\n'.format(bbox)) # REM
                    if a.get_clip_on():
                        clip_box = a.get_clip_box()
                        if clip_box is not None:
                            bbox = Bbox.intersection(bbox, clip_box)
                        clip_path = a.get_clip_path()
                        if clip_path is not None and bbox is not None:
                            clip_path = clip_path.get_fully_transformed_path()
                            bbox = Bbox.intersection(bbox,
                                                     clip_path.get_extents())
                    if bbox is not None and (bbox.width != 0 or
                                             bbox.height != 0):
                        bbox_filtered.append(bbox)

                print('35. bbox_filtered =\n{}\n'.format(bbox_filtered)) # REM
                if bbox_filtered:
                    _bbox = Bbox.union(bbox_filtered)
                    print('37. _bbox = {}\n'.format(_bbox)) # REM
                    trans = Affine2D().scale(1.0 / self.figure.dpi)
                    bbox_extra = TransformedBbox(_bbox, trans)
                    print('39. bbox_extra =\n{}\n'.format(bbox_extra)) # REM
                    bbox_inches = Bbox.union([bbox_inches, bbox_extra])
                    print('40. bbox_inches = {}\n'.format(bbox_inches)) # REM

                pad = kwargs.pop("pad_inches", None)
                if pad is None:
                    pad = rcParams['savefig.pad_inches']

                bbox_inches = bbox_inches.padded(pad)
                print('50. bbox_inches = {}\n'.format(bbox_inches)) # REM

            restore_bbox = tight_bbox.adjust_bbox(self.figure, bbox_inches,
                                                  canvas.fixed_dpi)
            print('60. restore_bbox = {}\n'.format(restore_bbox)) # REM

            _bbox_inches_restore = (bbox_inches, restore_bbox)
            print('70. _bbox_inches_restore = {}\n'.format(_bbox_inches_restore)) # REM
        else:
            _bbox_inches_restore = None

        try:
            result = print_method(
                filename,
                dpi=dpi,
                facecolor=facecolor,
                edgecolor=edgecolor,
                orientation=orientation,
                bbox_inches_restore=_bbox_inches_restore,
                **kwargs)
```

The bottom call fails. That's because `_bbox_inches_restore` is `nan`. I've added prints to see what's going on. Here's the output:
```
********************************************************************************
10. bbox_inches = tight

30. bbox_inches =
TransformedBbox(Bbox([[-2.626875, -9.760875], [418.22, 329.061375]]), Affine2D(array([[ 0.01388889,  0.        ,  0.        ],
       [ 0.        ,  0.01388889,  0.        ],
       [ 0.        ,  0.        ,  1.        ]])))

31. a = AxesSubplot(0.125,0.11;0.775x0.77)
32. bbox = Bbox(x0=54.1, y0=34.516, x1=418.22, y1=307.628)

31. a = Line2D(_line0)
32. bbox = Bbox(x0=198.067925666, y0=47.3226165927, x1=204.067925666, y1=53.3226165927)

31. a = Line2D((0.5,0),(0.5,0),...,(1e+07,0.0467085))
32. bbox = Bbox(x0=45.2147658927, y0=38.016, x1=345.6, y1=50.3226165927)

31. a = Line2D((34,0.0053605),(34,0.0053605),...,(1e+07,0.0467085))
32. bbox = Bbox(x0=114.609418301, y0=33.4283700922, x1=351.6, y1=56.3226165927)

31. a = Line2D(2-D)
32. bbox = Bbox(x0=inf, y0=inf, x1=-inf, y1=-inf)

31. a = Line2D(_line4)
32. bbox = Bbox(x0=115.500480178, y0=27.9487169682, x1=139.500480178, y1=51.9487169682)

31. a = Line2D(_line5)
32. bbox = Bbox(x0=193.8198322, y0=36.420110618, x1=199.8198322, y1=42.420110618)

31. a = Line2D((2,0),(2,0),...,(1e+07,0.00532915))
32. bbox = Bbox(x0=69.9852341073, y0=38.016, x1=345.6, y1=39.420110618)

31. a = Line2D((90.6667,0.00291536),(90.6667,0.00291536),...,(1e+07,0.00532915))
32. bbox = Bbox(x0=132.134989, y0=32.7841311028, x1=351.6, y1=45.420110618)

31. a = Line2D(3-D)
32. bbox = Bbox(x0=inf, y0=inf, x1=-inf, y1=-inf)

31. a = Line2D(_line9)
32. bbox = Bbox(x0=108.255582663, y0=26.3794168658, x1=132.255582663, y1=50.3794168658)

31. a = Line2D(_line10)
32. bbox = Bbox(x0=172.682163361, y0=35.4289737112, x1=178.682163361, y1=41.4289737112)

31. a = Line2D((1.4,0),(1.4,0),...,(1e+07,0.0015674))
32. bbox = Bbox(x0=63.6121248965, y0=38.016, x1=345.6, y1=38.4289737112)

31. a = Line2D((1.4,0.00015674),(1.4,0.00015674),...,(1e+07,0.0015674))
32. bbox = Bbox(x0=57.6121248965, y0=32.0572973711, x1=351.6, y1=44.4289737112)

31. a = Line2D(5-D)
32. bbox = Bbox(x0=inf, y0=inf, x1=-inf, y1=-inf)

31. a = Line2D(_line14)
32. bbox = Bbox(x0=99.1280912502, y0=26.0572973711, x1=123.12809125, y1=50.0572973711)

31. a = Line2D((1,0),(1,0),(4.7863e+08,0))
32. bbox = Bbox(x0=57.6, y0=38.016, x1=414.72, y1=38.016)

31. a = Line2D((1,0),(1,0),...,(1e+08,0))
32. bbox = Bbox(x0=51.6, y0=32.016, x1=392.742857143, y1=44.016)

31. a = Line2D(10-D)
32. bbox = Bbox(x0=inf, y0=inf, x1=-inf, y1=-inf)

31. a = Line2D((1,0),(1,0),(4.7863e+08,0))
32. bbox = Bbox(x0=57.6, y0=38.016, x1=414.72, y1=38.016)

31. a = Line2D((1,0),(1,0),...,(1e+08,0))
32. bbox = Bbox(x0=51.6, y0=32.016, x1=392.742857143, y1=44.016)

31. a = Line2D(20-D)
32. bbox = Bbox(x0=inf, y0=inf, x1=-inf, y1=-inf)

31. a = Line2D((1,0),(1,0),(4.7863e+08,0))
32. bbox = Bbox(x0=57.6, y0=38.016, x1=414.72, y1=38.016)

31. a = Line2D((1,0),(1,0),...,(1e+08,0))
32. bbox = Bbox(x0=51.6, y0=32.016, x1=392.742857143, y1=44.016)

31. a = Line2D(40-D)
32. bbox = Bbox(x0=inf, y0=inf, x1=-inf, y1=-inf)

31. a = Line2D(_line24)
32. bbox = Bbox(x0=339.6, y0=32.016, x1=383.28, y1=49.2855445545)

31. a = Line2D(_line25)
32. bbox = Bbox(x0=339.6, y0=32.016, x1=383.28, y1=99.8731722772)

31. a = Line2D(_line26)
32. bbox = Bbox(x0=339.6, y0=32.016, x1=383.28, y1=150.4608)

31. a = Line2D(_line27)
32. bbox = Bbox(x0=339.6, y0=32.4289737112, x1=383.28, y1=201.048427723)

31. a = Line2D(_line28)
32. bbox = Bbox(x0=339.6, y0=33.420110618, x1=383.28, y1=251.636055446)

31. a = Line2D(_line29)
32. bbox = Bbox(x0=339.6, y0=44.3226165927, x1=383.28, y1=302.223683168)

31. a = Line2D(_line30)
32. bbox = Bbox(x0=345.6, y0=38.016, x1=345.6, y1=301.493227723)

31. a = Text(8.12831e+07,0.02,u'40-D')
32. bbox = Bbox(x0=383.04, y0=36.0269508045, x1=419.08921875, y1=50.5441383045)

31. a = Text(8.12831e+07,0.212,u'20-D')
32. bbox = Bbox(x0=383.04, y0=86.6145785272, x1=419.08921875, y1=101.131766027)

31. a = Text(8.12831e+07,0.404,u'10-D')
32. bbox = Bbox(x0=383.04, y0=137.20220625, x1=419.08921875, y1=151.71939375)

31. a = Text(8.12831e+07,0.596,u'5-D')
32. bbox = Bbox(x0=383.04, y0=187.789833973, x1=409.54546875, y1=202.307021473)

31. a = Text(8.12831e+07,0.788,u'3-D')
32. bbox = Bbox(x0=383.04, y0=238.377461696, x1=409.54546875, y1=252.894649196)

31. a = Text(8.12831e+07,0.98,u'2-D')
32. bbox = Bbox(x0=383.04, y0=288.965089418, x1=409.54546875, y1=303.482276918)

31. a = Text(0.01,0.98,u'bbob-biobj - f1-f55\n58 targets in 1..-1.0e-4\n10 instances')
32. bbox = Bbox(x0=61.1712, y0=245.3708925, x1=253.622834375, y1=298.80576)

31. a = Text(0.5,0.01,u'v2.1.1, hv-hash=ff0e71e8cd978373')
32. bbox = Bbox(x0=145.7646875, y0=40.67712, x1=326.5553125, y1=50.355245)

31. a = Spine
32. bbox = Bbox(x0=57.6, y0=38.016, x1=57.6, y1=304.128)

31. a = Spine
32. bbox = Bbox(x0=414.72, y0=38.016, x1=414.72, y1=304.128)

31. a = Spine
32. bbox = Bbox(x0=57.6, y0=38.016, x1=414.72, y1=38.016)

31. a = Spine
32. bbox = Bbox(x0=57.6, y0=304.128, x1=414.72, y1=304.128)

31. a = XAxis(57.600000,38.016000)
32. bbox = Bbox(x0=0.0, y0=0.0, x1=0.0, y1=0.0)

31. a = YAxis(57.600000,38.016000)
32. bbox = Bbox(x0=0.0, y0=0.0, x1=0.0, y1=0.0)

31. a = Text(0.5,1,u'')
32. bbox = Bbox(x0=236.16, y0=310.128, x1=236.16, y1=310.128)

31. a = Text(0,1,u'')
32. bbox = Bbox(x0=57.6, y0=310.128, x1=57.6, y1=310.128)

31. a = Text(1,1,u'')
32. bbox = Bbox(x0=414.72, y0=310.128, x1=414.72, y1=310.128)

31. a = Rectangle(xy=(0, 0), width=1, height=1, angle=0)
32. bbox = Bbox(x0=57.6, y0=38.016, x1=414.72, y1=304.128)

35. bbox_filtered =
[Bbox([[54.1, 34.516], [418.22, 307.628]]), Bbox([[198.067925666, 47.3226165927], [204.067925666, 53.3226165927]]), Bbox([[45.2147658927, 38.016], [345.6, 50.3226165927]]), Bbox([[114.609418301, 33.4283700922], [351.6, 56.3226165927]]), Bbox([[inf, inf], [-inf, -inf]]), Bbox([[115.500480178, 38.016], [139.500480178, 51.9487169682]]), Bbox([[193.8198322, 36.420110618], [199.8198322, 42.420110618]]), Bbox([[69.9852341073, 38.016], [345.6, 39.420110618]]), Bbox([[132.134989, 32.7841311028], [351.6, 45.420110618]]), Bbox([[inf, inf], [-inf, -inf]]), Bbox([[108.255582663, 38.016], [132.255582663, 50.3794168658]]), Bbox([[172.682163361, 35.4289737112], [178.682163361, 41.4289737112]]), Bbox([[63.6121248965, 38.016], [345.6, 38.4289737112]]), Bbox([[57.6121248965, 32.0572973711], [351.6, 44.4289737112]]), Bbox([[inf, inf], [-inf, -inf]]), Bbox([[99.1280912502, 38.016], [123.12809125, 50.0572973711]]), Bbox([[57.6, 38.016], [414.72, 38.016]]), Bbox([[57.6, 38.016], [392.742857143, 44.016]]), Bbox([[57.6, 38.016], [414.72, 304.128]]), Bbox([[57.6, 38.016], [414.72, 38.016]]), Bbox([[57.6, 38.016], [392.742857143, 44.016]]), Bbox([[57.6, 38.016], [414.72, 304.128]]), Bbox([[57.6, 38.016], [414.72, 38.016]]), Bbox([[57.6, 38.016], [392.742857143, 44.016]]), Bbox([[57.6, 38.016], [414.72, 304.128]]), Bbox([[339.6, 32.016], [383.28, 49.2855445545]]), Bbox([[339.6, 32.016], [383.28, 99.8731722772]]), Bbox([[339.6, 32.016], [383.28, 150.4608]]), Bbox([[339.6, 32.4289737112], [383.28, 201.048427723]]), Bbox([[339.6, 33.420110618], [383.28, 251.636055446]]), Bbox([[339.6, 44.3226165927], [383.28, 302.223683168]]), Bbox([[345.6, 38.016], [345.6, 301.493227723]]), Bbox([[383.04, 36.0269508045], [419.08921875, 50.5441383045]]), Bbox([[383.04, 86.6145785272], [419.08921875, 101.131766027]]), Bbox([[383.04, 137.20220625], [419.08921875, 151.71939375]]), Bbox([[383.04, 187.789833973], [409.54546875, 202.307021473]]), Bbox([[383.04, 238.377461696], [409.54546875, 252.894649196]]), Bbox([[383.04, 288.965089418], [409.54546875, 303.482276918]]), Bbox([[61.1712, 245.3708925], [253.622834375, 298.80576]]), Bbox([[145.7646875, 40.67712], [326.5553125, 50.355245]]), Bbox([[57.6, 38.016], [57.6, 304.128]]), Bbox([[414.72, 38.016], [414.72, 304.128]]), Bbox([[57.6, 38.016], [414.72, 38.016]]), Bbox([[57.6, 304.128], [414.72, 304.128]]), Bbox([[57.6, 38.016], [414.72, 304.128]])]

37. _bbox = Bbox(x0=-inf, y0=-inf, x1=inf, y1=inf)

39. bbox_extra =
TransformedBbox(Bbox([[-inf, -inf], [inf, inf]]), Affine2D(array([[ 0.01388889,  0.        ,  0.        ],
       [ 0.        ,  0.01388889,  0.        ],
       [ 0.        ,  0.        ,  1.        ]])))

40. bbox_inches = Bbox(x0=nan, y0=nan, x1=nan, y1=nan)

50. bbox_inches = Bbox(x0=nan, y0=nan, x1=nan, y1=nan)

60. restore_bbox = <function restore_bbox at 0x000000000A664358>

70. _bbox_inches_restore = (Bbox([[nan, nan], [nan, nan]]), <function restore_bbox at 0x000000000A664358>)
```

Clearly matplotlib cannot handle `inf` values in bounding boxes, the `union()` method then returns `nan`. Upon further investigation, it turns out that the artists that have `inf` in their bounding boxes are those labeled `2-D`, `3-D` and so on. These are the 'label' plots produced by the following code in `ppfig.py`, line 533.
```
    if 'label' in kwargs:
        res3 = plt.plot([np.nan], [np.nan], **kwargs)
        for i in res3:
            i.update_from(res[0])  # copy all attributes of res
        res.extend(res3)
```

**Comment by islandblue** [comment]
So a quick go at a fix is to do here what we recently did with the `development` branch. Pass `[0.]` instead of `[np.nan]` as coordinates, but adjust `kwargs` so nothing is drawn, as follows.
```
    if 'label' in kwargs:
        res3 = plt.plot([0.], [0.], **kwargs)
        for i in res3:
            i.update_from(res[0])  # copy all attributes of res
        plt.setp(res3, linestyle='', marker='')
        res.extend(res3)
```

A quick test did run through without problems. @liujianxun-ict, you can patch this one manually on your code. Since a release is well on its way, we may opt to not push to master at this time.

**Comment by islandblue** [comment]
This will be closed with #1724
